src/groups/general_linear.jl

```diff
@@ -0,0 +1,241 @@
+    expX = exp(X)
+    if isnormal(X; atol=sqrt(eps(real(eltype(X)))))
+        return compose!(G, q, p, expX)
+    end
+    compose!(G, q, expX', exp(X - X'))
+    compose!(G, q, p, q)
+    return q
+end
+function exp!(::GeneralLinear{1}, q, p, X)
+    p1 = p isa Identity ? p : p[1]
+    q[1] = p1 * exp(X[1])
+    return q
+end
+function exp!(G::GeneralLinear{2}, q, p, X)
+    if isnormal(X; atol=sqrt(eps(real(eltype(X)))))
+        return compose!(G, q, p, exp(SizedMatrix{2,2}(X)))
+    end
+    A = SizedMatrix{2,2}(X')
+    B = SizedMatrix{2,2}(X) - A
+    compose!(G, q, exp(A), exp(B))
+    compose!(G, q, p, q)
+    return q
+end
+
+flat!(::GeneralLinear, ξ::CoTFVector, p, X::TFVector) = copyto!(ξ, X)
+
+get_coordinates(::GeneralLinear{n,ℝ}, p, X, ::DefaultOrthonormalBasis) where {n} = vec(X)
+
+function get_coordinates!(
+    ::GeneralLinear{n,ℝ},
+    Xⁱ,
+    p,
+    X,
+    ::DefaultOrthonormalBasis,
+) where {n}
+    return copyto!(Xⁱ, X)
+end
+
+function get_vector(::GeneralLinear{n,ℝ}, p, Xⁱ, ::DefaultOrthonormalBasis) where {n}
+    return reshape(Xⁱ, n, n)
+end
+
+function get_vector!(::GeneralLinear{n,ℝ}, X, p, Xⁱ, ::DefaultOrthonormalBasis) where {n}
+    return copyto!(X, Xⁱ)
+end
+
+function group_exp!(::GeneralLinear{1}, q, X)
+    q[1] = exp(X[1])
+    return q
+end
+group_exp!(::GeneralLinear{2}, q, X) = copyto!(q, exp(SizedMatrix{2,2}(X)))
+
+function group_log!(::GeneralLinear{1}, X::AbstractMatrix, p::AbstractMatrix)
+    X[1] = log(p[1])
+    return X
+end
+
+inner(::GeneralLinear, p, X, Y) = dot(X, Y)
+
+invariant_metric_dispatch(::GeneralLinear, ::LeftAction) = Val(true)
+
+inverse_translate_diff(::GeneralLinear, p, q, X, ::LeftAction) = X
+inverse_translate_diff(::GeneralLinear, p, q, X, ::RightAction) = p * X / p
+
+function inverse_translate_diff!(G::GeneralLinear, Y, p, q, X, conv::ActionDirection)
+    return copyto!(Y, inverse_translate_diff(G, p, q, X, conv))
+end
+
+# find sU for s ∈ S⁺ and U ∈ U(n, 𝔽) that minimizes ‖sU - p‖²
+function _project_Un_S⁺(p)
+    n = LinearAlgebra.checksquare(p)
+    F = svd(p)
+    s = mean(F.S)
+    U = F.U * F.Vt
+    return rmul!(U, s)
+end
+
+@doc raw"""
+    log(G::GeneralLinear, p, q)
+
+Compute the logarithmic map on the [`GeneralLinear(n)`](@ref) group.
+
+The algorithm proceeds in two stages. First, the point $r = p^{-1} q$ is projected to the
+nearest element (under the Frobenius norm) of the direct product subgroup
+$\mathrm{O}(n) × S^+$, whose logarithmic map is exactly computed using the matrix logarithm.
+This initial tangent vector is then refined using the  [`NLsolveInverseRetraction`](@ref).
+
+For `GeneralLinear(n, ℂ)`, the logarithmic map is instead computed on the realified
+supergroup `GeneralLinear(2n)` and the resulting tangent vector is then complexified.
+
+Note that this implementation is experimental.
+"""
+log(::GeneralLinear, p, q)
+
+# TODO: Look into using geodesic shooting method in §6.2 of http://hdl.handle.net/2078.1/132587
```

Comment:
Should we keep a todo in our code? Should this maybe become an issue to keep better track of it?

Reply:
Yeah, perhaps an issue would be more appropriate. The goal here is that a user/developer reading the code understands that there are alternative approaches that might be better. Unless a user browses our issues, they may not find it. A compromise would be to create an issue and have a comment linking to the issue in the code.

Reply:
I moved it to an issue and opted to remove the comment. Also, the point in the comment about normal matrices was incorrect. It is not the case I believe that the exponential of a normal matrix is normal. And many normal matrices will be invalid for our purposes. e.g. we need the projected matrix to also have a positive determinant and be non-singular, which is not guaranteed.